toolchain/check/diagnostic_emitter.cpp

```diff
@@ -164,6 +164,22 @@ auto DiagnosticEmitter::ConvertArg(llvm::Any arg) const -> llvm::Any {
     return llvm::APSInt(typed_int->value,
                         !sem_ir_->types().IsSignedInt(typed_int->type));
   }
+  if (auto* specific_interface =
+          llvm::any_cast<SemIR::SpecificInterface>(&arg)) {
+    return "`" + StringifySpecificInterface(*sem_ir_, *specific_interface) +
+           "`";
+  }
+  if (auto* specific_interface_id =
+          llvm::any_cast<SemIR::SpecificInterfaceId>(&arg)) {
+    auto specific_interface =
+        sem_ir_->specific_interfaces().Get(*specific_interface_id);
+    return "`" + StringifySpecificInterface(*sem_ir_, specific_interface) + "`";
+  }
+  if (auto* impl_id = llvm::any_cast<SemIR::ImplId>(&arg)) {
+    const auto& impl = sem_ir_->impls().Get(*impl_id);
+    return "`" + StringifyConstantInst(*sem_ir_, impl.self_id) + " as " +
+           StringifySpecificInterface(*sem_ir_, impl.interface) + "`";
+  }
```

Comment:
I think this is unused and untested; maybe drop it for now?

Reply:
Done